### PR TITLE
Terror Spider Fixes & Tweaks

### LINKED ISSUE #11506: Giant spiders ruined by the AI change
**Problem Description**:
As of the recent changes, giant spiders (and AI terrors too) are unable to perform their abilities if there are no players nearby. As a result of all hostile mobs having their AI disabled if there are no players around, giant spiders do not make webs, wrap players or random objects anymore, and since the moment they see someone they'll chase after him, they basically won't ever have time to do their automatic actions.

**What did you expect to happen**: 
Giant spiders should move around, make webs and wrap items when they are on their own. That is how they "get out of hand", but currently they just stand still wherever they grew till a player walks in their direction.

**What happened instead**:
Giant spiders, like every other hostile mob, don't move / do anything until a player is in their view.

**Why is this bad/What are the consequences:**
Renders giant spiders useless as they can't wrap things to make more eggs unless a live player is around, and if the player is of a different faction, the spiders will chase after it, and once it's gone or dead, they'll go back to standing still.

**Steps to reproduce the problem**:
Observe giant spiders, specifically, nurses
See how they don't move or wrap items around them due to their AI status being set to AI_Z_OFF
Spawn in a corpse
No response from them

**When did the problem start happening**:
If your report is about something that used to work but no longer does, when was the last time you remember it working?
Everything worked fine until this PR got merged https://github.com/ParadiseSS13/Paradise/pull/11450

### LINKED ISSUE #11782: Terror Spider Gateway
**Problem Description**:
I don't know how to exactly explain it. But it seems like the gateway doesn't properly start itself. The Queen spider does not lay eggs, the death of the white spider does not open the command blast doors and the rest of the spiders don't lay webs either.

**What did you expect to happen**: 
That the gateway isn't a gigantic snore fest that can only be completed with an emag.

**What happened instead**:
Once the initial spiders are dead, there are no threats leftover. Making it pretty much the empty station gateway.

**Why is this bad/What are the consequences:**
The Terror Spider Gateway is the only gateway left that's worth doing. Since it's unlikely we'll get Wild West back at this point.

**Steps to reproduce the problem**:
Start a round with the Terror Spider Gateway, look at the spiders and see no webs, no eggs, nothing.

**When did the problem start happening**:
I think it happened after the lavaland merge. I don't know for sure.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was anything else out of the ordinary happening?)**:
spooders

### PR DESCRIPTION
**Pull request by Kyep**
Fixes #11782 
Fixes #11506 

:cl: Kyep
fix: Terror spider away mission is no longer unplayable due to lack of spiders spawning
fix: Terror spiders no longer smash random things for no reason
fix: Gray terror spiders now use the correct web type
tweak: Tweaked gateway terror spider behavior a bit to account for the fact their AI is now only active when living players are in the gateway
refactor: Cleaned up some terror spider code
fix: White terror spiders (and their webs) are no longer able to infect slimes.
fix: Terror spiders in general no longer allow you to control them if you've rendered yourself unrespawnable
tweak: Reworked gray terrors. Previously they (according to class role, and AI coding) hid in vents and ambushed people who got close or walked into vents. Players did not play this way, though, so now they've been changed to simply be stealthed (like xeno hunters) whenever they're on a web. Their AI has also been dramatically simplified. To balance this ability, they've lost the fast health regen they used to have.
fix: Fixed case of terror spider attack animations not working (AI spider attacking light, prince using special ability)
fix: Fixed white terror infections not properly causing confusion
/:cl:

### CODE REVIEW

**Comment by Emanthealmighty** [comment]
Could this be applied to giant spiders too?

**Comment by Kyep** [comment]
Yes, but if we're wanting to apply it to other mobs, I might change the approach. Instead of overriding the proc on /terrorspider, instead I'd just have terrorspider + nurse regular spider + whatever else have a var like 'aiZLevelCheck' or something which is set to TRUE for those mobs that are always active whenever a player is on their zlevel.

**Comment by Emanthealmighty** [comment]
I'd say that having a general var like the one you described would be ideal.

**Comment by Kyep** [comment]
@Emanthealmighty changed, it now affects nurse spiders of the giant spiders (but not the hunter/guard giant spider)

**Comment by Emanthealmighty** [comment]
> @Emanthealmighty it now affects giant nurse spiders (the green-eyed giant spiders that lay eggs) too. It still doesn't affect the hunter/guard giant spider, as I don't think the effects on them would be too useful. There may be other bugs with giant spiders that stop them wrapping things, but at least with this fix, it won't be the zlevel/distance stuff.
> 
> I'm still not happy with how sluggish gateway terror spiders are to respond when a player gets in range of them... but I am not sure how to change that (or even if it is changable without having them scan much more often for hostiles).
> 
> Still, they're once again populating the mission, and they do respond (eventually) when you get in range of them. So, the mission is playable again.

I think making them scan more often would be a solution. Before the gateway mission broke, I noticed they don't even wrap up bodies that often, while surprisingly, giant spiders did.

**Comment by Kyep** [comment]
@Emanthealmighty Making them scan more often would be a solution, one that I might implement in future. Wrapping bodies is meant to be a low-frequency behavior as they get little benefit from it and there are unlikely to be bodies in the gateway for them to wrap.

**Comment by Kyep** [comment]
By 'in future' I mean 'in another PR'. I think this PR is ready as-is.

**Comment by Kyep** [comment]
Test

**Comment by Kyep** [comment]
This PR has been rewritten. Major changes:
- Much more focus on code quality (it now includes some related refactors and cleanup, it no longer overrides consider_wakeup, etc)
- It now works better for giant spiders as well
Description updated.

**Comment by Kyep** [comment]
More fixes added:
fix: White terror spiders (and their webs) are no longer able to infect slimes.
fix: Terror spiders in general no longer allow you to control them if you've rendered yourself unrespawnable

**Comment by Kyep** [comment]
Added:
tweak: Reworked gray terrors. Previously they (according to class role, and AI coding) hid in vents and ambushed people who got close or walked into vents. Players did not play this way, though, so now they've been changed to simply be stealthed (like xeno hunters) whenever they're on a web. Their AI has also been dramatically simplified.

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_ai.dm [hunk 1]
```diff
@@ -92,10 +90,18 @@
...
-		return ..()
-	if(AIStatus != AI_OFF && !target)
+	if(target)
+		CreatePath(target)
+	return ..()
+
+/mob/living/simple_animal/hostile/poison/terror_spider/handle_automated_movement()
+	// Putting the main terror spider AI code in handle_automated_movement() rather than handle_automated_action() ensures it will still run when the spider AIStatus == AI_IDLE
+	// This is necessary for the terror spiders in the away mission to work properly.
+	if(stat || ckey || AIStatus != AI_IDLE)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_ai.dm]: there's no way that `handle_automated_movement` gets called on something with that's dead or has a ckey.

Comment (Kyep) [reply code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_ai.dm]: Updated to remove those two checks.

**Comment by Kyep** [comment]
Updated to fix merge conflict and remove stuff related to giant spiders that was changed in Fox's PR.

**Comment by Kyep** [comment]
pdated to fix merge conflict and remove stuff related to giant spiders that was changed in Fox's PR.

**Pull request merged**